datalad_next/annexremote/tests/test_uncurl.py

```diff
@@ -0,0 +1,269 @@
+    baseurl = (tmp_path / "ria").as_uri()
+    srcds.create_sibling_ria(
+        # use a ria+file:// URL for simplicity
+        f'ria+{baseurl}',
+        name='ria',
+        new_store_ok=True,
+        **res_kwargs
+    )
+    srcds.push(to='ria', **res_kwargs)
+    # setup is done
+
+    # start of the actual test
+    # create a fresh dataset
+    ds = EnsureDataset()(tmp_path / 'testds').ds.create(**res_kwargs)
+    dsca = ds.repo.call_annex
+    # we add uncurl WITH NO config whatsoever.
+    # this must be enough to be able to use the built-in downloaders
+    target_fname = 'mydownload.txt'
+    dsca(['initremote', 'myuncurl'] + std_initargs)
+    dsca(['addurl', '--file', target_fname,
+          # we download from the verbatim, hand-crafted URL
+          f'{baseurl}/{srcds.id[:3]}/{srcds.id[3:]}/annex/objects/'
+          f'{testkey_props["hashdirmixed"]}'
+          f'{testkey_props["key"]}/{testkey_props["key"]}'
+    ])
+    assert (ds.pathobj / target_fname).read_text() == testfile_content
+    # make sure the re-downloaded key ends up having the same keyname in
+    # the new dataset
+    assert ds.status(
+        target_fname, annex='basic', return_type='item-or-list',
+        **res_kwargs)['key'] == testkey_props['key']
+
+    # now we drop the key...
+    ds.drop(target_fname, **res_kwargs)
+    assert not (ds.pathobj / target_fname).exists()
+    # ...and we move the RIA store to break the recorded
+    # URL (simulating an infrastructure change)
+    (tmp_path / 'ria').rename(tmp_path / 'ria_moved')
+
+    # verify that no residual magic makes data access possible
+    with pytest.raises(IncompleteResultsError):
+        ds.get(target_fname, **res_kwargs)
+
+    # fix it via an access URL config,
+    # point directly via a hard-coded dataset ID
+    # NOTE: last line is no f-string!
+    url_tmpl = f'{(tmp_path / "ria_moved").as_uri()}/{srcds.id[:3]}/{srcds.id[3:]}' \
+        + '/annex/objects/{annex_dirhash}/{annex_key}/{annex_key}'
+    ds.configuration(
+        'set', f'remote.myuncurl.uncurl-url={url_tmpl}', **res_kwargs)
+    # confirm checkpresent acknowledges this
+    dsca(['fsck', '-q', '-f', 'myuncurl'])
+    # confirm transfer_retrieve acknowledges this
+    ds.get(target_fname, **res_kwargs)
+
+    # but we can also do without hard-coding anything, so let's drop again
+    ds.drop(target_fname, **res_kwargs)
+    assert not (ds.pathobj / target_fname).exists()
+
+    # for that we need to add a match expression that can "understand"
+    # the original URL. All we need is to distinguish the old base path
+    # from the structured components in the RIA store (for the
+    # latter we can simple account for 4 levels of sudirs
+    ds.configuration(
+        'set',
+        'remote.myuncurl.uncurl-match='
+        'file://(?P<basepath>.*)/(?P<dsdir>[^/]+/[^/]+)/annex/objects/.*$',
+        **res_kwargs)
+    # NOTE: last line is no f-string!
+    url_tmpl = f'{(tmp_path / "ria_moved").as_uri()}' \
```

Comment:
Same here. I may miss something, but I think the f-string is superfluous.
```suggestion
    url_tmpl = (tmp_path / "ria_moved").as_uri()' \
```

Reply:
Thx, I removed it. I could not commit the suggested change directly, due to the syntax error.